cli/internal/providers/event-stream/source/handler.go

```diff
@@ -396,12 +396,17 @@ func (p *Handler) LoadStateFromResources(ctx context.Context, collection *resour
 		if !ok {
 			return nil, fmt.Errorf("unable to cast resource to event stream source")
 		}
-		var trackingPlanURN string
-		var err error
+		var trackingPlanURN *string
 		if source.TrackingPlan != nil {
-			trackingPlanURN, err = collection.GetURNByID(dcstate.TrackingPlanResourceType, source.TrackingPlan.ID)
-			if err != nil {
+			tpURN, err := collection.GetURNByID(dcstate.TrackingPlanResourceType, source.TrackingPlan.ID)
+			if err == resources.ErrRemoteResourceNotFound {
```

Comment:
@shekhar-rudder the resource would be there in the resource collection but the resource would not have an externalId, we are adding all resources to the resource collection(with or without externalId)

Reply:
we load all resources, the same map is used for import flow and state recon